packages/loopring_v3/contracts/iface/ExchangeData.sol

```diff
@@ -91,60 +85,86 @@ library ExchangeData
+    // at the time the request is being made) the full balance will be withdrawn.
+    struct ForcedWithdrawal
+    {
+        address owner;
+        uint32  timestamp;
+        uint64  fee;
     }
 
     struct Constants
     {
         uint SNARK_SCALAR_FIELD;
-        uint MAX_OPEN_DEPOSIT_REQUESTS;
-        uint MAX_OPEN_WITHDRAWAL_REQUESTS;
-        uint MAX_AGE_REQUEST_UNTIL_FORCED;
-        uint MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE;
-        uint MAX_TIME_IN_SHUTDOWN_BASE;
-        uint MAX_TIME_IN_SHUTDOWN_DELTA;
+        uint MAX_OPEN_FORCED_REQUESTS;
+        uint MAX_AGE_FORCED_REQUEST_UNTIL_WITHDRAW_MODE;
         uint TIMESTAMP_HALF_WINDOW_SIZE_IN_SECONDS;
         uint MAX_NUM_TOKENS;
-        uint MAX_NUM_ACCOUNTS;
-        uint FEE_BLOCK_FINE_START_TIME;
-        uint FEE_BLOCK_FINE_MAX_DURATION;
         uint MIN_AGE_PROTOCOL_FEES_UNTIL_UPDATED;
-        uint GAS_LIMIT_SEND_TOKENS;
+        uint MIN_TIME_IN_SHUTDOWN;
+        uint TX_DATA_AVAILABILITY_SIZE;
+        uint MAX_AGE_DEPOSIT_UNTIL_WITHDRAWABLE;
     }
 
     function SNARK_SCALAR_FIELD() internal pure returns (uint) {
         // This is the prime number that is used for the alt_bn128 elliptic curve, see EIP-196.
         return 21888242871839275222246405745257275088548364400416034343698204186575808495617;
     }
-
-    function MAX_OPEN_DEPOSIT_REQUESTS() internal pure returns (uint16) { return 1024; }
-    function MAX_OPEN_WITHDRAWAL_REQUESTS() internal pure returns (uint16) { return 1024; }
-    function MAX_AGE_REQUEST_UNTIL_FORCED() internal pure returns (uint32) { return 14 days; }
-    function MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 15 days; }
-    function MAX_TIME_IN_SHUTDOWN_BASE() internal pure returns (uint32) { return 30 days; }
-    function MAX_TIME_IN_SHUTDOWN_DELTA() internal pure returns (uint32) { return 1 seconds; }
+    function MAX_OPEN_FORCED_REQUESTS() internal pure returns (uint16) { return 1024; }
+    function MAX_AGE_FORCED_REQUEST_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 15 days; }
     function TIMESTAMP_HALF_WINDOW_SIZE_IN_SECONDS() internal pure returns (uint32) { return 7 days; }
-    function MAX_NUM_TOKENS() internal pure returns (uint) { return 2 ** 10; }
-    function MAX_NUM_ACCOUNTS() internal pure returns (uint) { return 2 ** 24 - 1; }
-    function FEE_BLOCK_FINE_START_TIME() internal pure returns (uint32) { return 6 hours; }
-    function FEE_BLOCK_FINE_MAX_DURATION() internal pure returns (uint32) { return 6 hours; }
+    function MAX_NUM_TOKENS() internal pure returns (uint) { return 2 ** 12; }
     function MIN_AGE_PROTOCOL_FEES_UNTIL_UPDATED() internal pure returns (uint32) { return 1 days; }
-    function GAS_LIMIT_SEND_TOKENS() internal pure returns (uint32) { return 80000; }
+    function MIN_TIME_IN_SHUTDOWN() internal pure returns (uint32) { return 28 days; }
+    function TX_DATA_AVAILABILITY_SIZE() internal pure returns (uint32) { return 104; }
```

Comment:
It isn't actually, I actually first tried to keep it a power of 2 and then settled for 100, but I needed some extra bytes. :)
So it's really just the max amount of bytes needed for all different transaction types.

I do want to keep it < 128 for the data compression though (e.g. 1 bit for a mode zero/non-zero and 7 bits for length of zeros/actual data, only using a single byte overhead), but it's not like that's even that important. :)

Reply:
It deserves some comments in the file.

Reply:
> Should we rename it to MAX_TX_DATA_AVAILABILITY_SIZE?

It's the size of each transaction because the data is padded with zeros when necessary to get to this size.